File: src/main/java/dev/fringe/crud/config/ContextDataSource.java
package dev.fringe.crud.config;

import org.apache.tomcat.dbcp.dbcp2.BasicDataSource;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;
import javax.sql.DataSource;

@Configuration
public class ContextDataSource {
    @Bean
    public DataSource dataSource() {
        BasicDataSource dataSource = new BasicDataSource();
        dataSource.setDriverClassName("org.hsqldb.jdbcDriver");
        dataSource.setUrl("jdbc:hsqldb:mem:fringe");
        dataSource.setUsername("sa");
        dataSource.setPassword("");
        return dataSource;
    }
}

File: src/main/scala/dev/fringe/crud/config/ContextDataSource.scala
package dev.fringe.crud.config

import javax.sql.DataSource

import org.apache.tomcat.dbcp.dbcp2.BasicDataSource
import org.springframework.context.annotation.{Bean, Configuration}

@Configuration
class ContextDataSource {

  @Bean def dataSource: DataSource = {
    val dataSource = new BasicDataSource
    dataSource.setDriverClassName("org.hsqldb.jdbcDriver")
    dataSource.setUrl("jdbc:hsqldb:mem:fringe")
    dataSource.setUsername("sa")
    dataSource.setPassword("")
    dataSource
  }
}
